crates/network-scanner-net/src/runtime.rs

```diff
@@ -106,14 +106,18 @@ impl Socket2Runtime {
                 let mut events_happened = HashMap::new();
 
                 loop {
-                    if is_terminated.load(Ordering::Acquire) {
+                    let Some(runtime) = runtime.upgrade() else {
+                        break;
+                    };
+
+                    if runtime.is_terminated.load(Ordering::Acquire) {
                         break;
                     }
 
                     tracing::debug!("polling events");
-                    if let Err(e) = poller.wait(&mut events, None) {
+                    if let Err(e) = runtime.poller.wait(&mut events, None) {
                         tracing::error!(error = ?e, "failed to poll events");
-                        is_terminated.store(true, Ordering::SeqCst);
+                        runtime.is_terminated.store(true, Ordering::SeqCst);
```

Comment:
would it panic if runtime is dropped here? as poller.wait() will stop blocking because of notify in the drop function of runtime?